src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -14,10 +14,89 @@
 
 package codeu.controller;
 
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 /** Servlet class responsible for the chat page. */
 public class AdminServlet extends HttpServlet {
-  // TODO(JW): To be implemented.
 
+  /** Store class that gives access to Conversations. */
+  private static ConversationStore conversationStore;
+
+  /** Store class that gives access to Messages. */
+  private static MessageStore messageStore;
+
+  /** Store class that gives access to Users. */
+  private static UserStore userStore;
+
+  /** Set up state for handling data collection for the Admin Page. */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setUserStore(UserStore.getInstance());
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    String username = (String) request.getSession().getAttribute("user");
+    Boolean admin = userStore.getUser(username).isAdmin();
+    if (!admin) {
+      response.sendRedirect("/index");
+      return;
+    }
+  }
```

Comment:
I think you need a `response.sendRedirect` call at the end of `doPost`.

Reply:
Indeed, fixed